csharp/src/Ice/SslTransceiver.cs

```diff
@@ -552,32 +552,12 @@ private void FinishAuthenticate()
                 bool certificateNameMismatch = (errors & (int)SslPolicyErrors.RemoteCertificateNameMismatch) > 0;
                 if (certificateNameMismatch)
                 {
-                    if (_engine.CheckCertName && !string.IsNullOrEmpty(_host))
-                    {
-                        if (_engine.SecurityTraceLevel >= 1)
-                        {
-                            string msg = "SSL certificate validation failed - Hostname mismatch";
-                            if (_verifyPeer == 0)
-                            {
-                                msg += " (ignored)";
-                            }
-                            _communicator.Logger.Trace(_engine.SecurityTraceCategory, msg);
-                        }
-
-                        if (_verifyPeer > 0)
-                        {
-                            return false;
-                        }
-                        else
-                        {
-                            errors ^= (int)SslPolicyErrors.RemoteCertificateNameMismatch;
-                        }
-                    }
-                    else
+                    if (_engine.SecurityTraceLevel >= 1)
                     {
-                        errors ^= (int)SslPolicyErrors.RemoteCertificateNameMismatch;
-                        certificateNameMismatch = false;
+                        _communicator.Logger.Trace(_engine.SecurityTraceCategory,
+                            "SSL certificate validation failed - Hostname mismatch");
                     }
+                    return false;
```

Comment:
Why bypass the rest of the error checks?

Reply:
There is a separate PR for VerifyPeer see #899